### PR TITLE
xds-k8s: handle missing edge case in TestConfig version comparison

### PR DESCRIPTION
**Pull request by sergiitk**
Fixes incorrect `master` version handling in subsetting_test.

Ref b/235825277

### CODE REVIEW

**Comment by sergiitk** [comment]
- [grpc/core/master/linux/psm-security](https://fusion2.corp.google.com/invocations/feca317d-61ab-4ee2-a099-c53588a9b87f)
- [grpc/core/master/linux/grpc_xds_k8s_lb](https://fusion2.corp.google.com/invocations/e86606a4-c4ba-4777-a76c-e41faf47b1c4)
- [grpc/core/master/linux/grpc_xds_url_map](https://fusion2.corp.google.com/invocations/811969e6-e6a6-4f76-a595-70c8661314b4)

**Comment by sergiitk** [comment]
[grpc/core/v1.47.x/branch/linux/grpc_xds_k8s_lb](https://source.cloud.google.com/results/invocations/6d758a0c-31e7-4ea2-9908-dac65c43452c)

**Comment by sergiitk** [comment]
Confirming the fix works:

```
skips.py:113] Skipping TestConfig(client_lang='cpp', server_lang='cpp', version='v1.47.x')
[  SKIPPED ] setUpClass (__main__.SubsettingTest)
----------------------------------------------------------------------
Ran 0 tests in 0.001s

OK (skipped=1)
```

File: tools/run_tests/xds_k8s_test_driver/framework/helpers/skips.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def version_gte(self, another: str) -> bool:
         """
         if self.version == 'master' or self.version is None:
             return True
+        if another == 'master':
+            return False
         return self._parse_version(self.version) >= self._parse_version(another)
 
     def version_lt(self, another: str) -> bool:
```
Comment (gnossen) [inline tools/run_tests/xds_k8s_test_driver/framework/helpers/skips.py:72]: Nit: Can we just remove one of `version_gte` and `version_lt`. Can be done in a separate PR.

**Review by gnossen — APPROVED** [review]

**Pull request merged**